py/desiutil/plots.py

```diff
@@ -125,26 +125,60 @@ def plot_slices(x, y, x_lo, x_hi, y_cut, num_slices=5, min_count=100, axis=None,
 
 
 class MaskedArrayWithLimits(numpy.ma.MaskedArray):
-    """Masked array with additional vmin, vmax attributes.
+    """Masked array with additional `vmin`, `vmax` attributes.
 
-    Based on https://docs.scipy.org/doc/numpy/user/
-    basics.subclassing.html#simple-example-adding-an-extra-attribute-to-ndarray
+    This class accepts the same arguments as
+    :class:`~numpy.ma.MaskedArray`.
+
+    Based on https://docs.scipy.org/doc/numpy/user/basics.subclassing.html#simple-example-adding-an-extra-attribute-to-ndarray
 
     This is not a general-purpose subclass and is only intended to simplify
-    passing vmin, vmax limits from prepare_data() to the plotting utility
-    methods defined in this module.
+    passing `vmin`, `vmax` limits from :func:`~desiutil.plots.prepare_data` to
+    the plotting utility methods defined in this module.
+
+    Parameters
+    ----------
+    vmin : :class:`float`, optional
+        Minimum value when used for clipping or masking.
+    vmax : :class:`float`, optional
+        Maximum value when used for clipping or masking.
+
+    Attributes
+    ----------
+    vmin : :class:`float`
+        Minimum value when used for clipping or masking.
+    vmax : :class:`float`
+        Maximum value when used for clipping or masking.
     """
-    def __new__(cls, data, mask, dtype=None, order=None, vmin=None, vmax=None):
-        obj = numpy.ma.MaskedArray.__new__(cls, data, mask)
-        obj.vmin = vmin
-        obj.vmax = vmax
+    def __new__(cls, *args, **kwargs):
+        obj = super(MaskedArrayWithLimits, cls).__new__(cls, *args, **kwargs)
+        if 'vmin' in kwargs:
+            obj._optinfo['vmin'] = kwargs['vmin']
+        #     obj.vmin = kwargs['vmin']
+        # else:
+        #     obj.vmin = None
+        if 'vmax' in kwargs:
+            obj._optinfo['vmax'] = kwargs['vmax']
+        #     obj.vmax = kwargs['vmax']
+        # else:
+        #     obj.vmax = None
         return obj
 
-    def __array_finalize__(self, obj):
-        if obj is None: return
-        numpy.ma.MaskedArray.__array_finalize__(self, obj)
-        self.vmin = getattr(obj, 'vmin', None)
-        self.vmax = getattr(obj, 'vmax', None)
+    # def __array_finalize__(self, obj):
```

Comment:
I'm not sure what a "dunder" method is, but I do know that the original code was *not* working as expected, and the new code is.  Please see the unit tests.